### PR TITLE
feature-NXP-20838-subscribe-unsubscribe-operations

### PR DESCRIPTION
**Pull request by mnixo**
Added operations to subscribe and unsubscribe document notifications. It's possible to subscribe all available notifications or specify notifications individually.

### CODE REVIEW

File: nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/automation/SubscribeOperation.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import org.nuxeo.ecm.platform.ec.notification.NotificationConstants;
+import org.nuxeo.ecm.platform.notification.api.NotificationManager;
+
+/**
+ * Operation to subscribe a user to the notifications of one or more of documents.
+ *
+ * @since 8.10
+ */
+@Operation(id = SubscribeOperation.ID, category = Constants.CAT_DOCUMENT, label = "Subscribe document",
+    description = "Subscribe one or more documents. No value is returned.", aliases = { "Collection.Subscribe" })
```
Comment (efge) [inline nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/automation/SubscribeOperation.java]: Why the alias? Is it needed? I'm not sure it helps, we usually only have aliases for backward compatibility.

**Comment by efge** [comment]
👍

File: nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/automation/SubscribeOperation.java (hunk 1 above)
Comment (mnixo) [inline nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/automation/SubscribeOperation.java]: Pretty sure it isn't needed. I added it to be as consistent as possible with [AddToFavoritesOperation](https://github.com/nuxeo/nuxeo/blob/8032735a06247583495e15f7a082bcf7cabdd2aa/nuxeo-features/nuxeo-platform-collections/nuxeo-platform-collections-core/src/main/java/org/nuxeo/ecm/collections/core/automation/AddToFavoritesOperation.java#L36) and [RemoveFromFavoritesOperation](https://github.com/nuxeo/nuxeo/blob/8032735a06247583495e15f7a082bcf7cabdd2aa/nuxeo-features/nuxeo-platform-collections/nuxeo-platform-collections-core/src/main/java/org/nuxeo/ecm/collections/core/automation/RemoveFromFavoritesOperation.java#L36). But I'll take the aliases from these asap.

**Comment by atchertchian** [comment]
:+1:

**Pull request closed without merging**